src/components/application_manager/test/hmi_language_handler_test.cc

```diff
@@ -0,0 +1,378 @@
+#include "resumption/last_state_impl.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+#include "utils/lock.h"
+
+namespace test {
+namespace components {
+namespace hmi_language_handler {
+
+namespace am = ::application_manager;
+
+using am::event_engine::Event;
+using am::ApplicationSet;
+using ::utils::SharedPtr;
+
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
+using ::testing::_;
+
+typedef NiceMock<
+    ::test::components::application_manager_test::MockApplicationManager>
+    MockApplicationManager;
+typedef NiceMock<
+    ::test::components::application_manager_test::MockHMICapabilities>
+    MockHMICapabilities;
+typedef NiceMock< ::test::components::event_engine_test::MockEventDispatcher>
+    MockEventDispatcher;
+typedef NiceMock<application_manager_test::MockApplication> MockApp;
+
+typedef SharedPtr<MockApp> ApplicationSharedPtr;
+typedef am::HMILanguageHandler::Apps Apps;
+
+namespace {
+const uint32_t kDefaultAppsSize = 0u;
+}  // namespace
+
+class HmiLanguageHandlerTest : public ::testing::Test {
+ public:
+  HmiLanguageHandlerTest()
+      : last_state_("app_storage_folder", "app_info_storage") {
+    EXPECT_CALL(app_manager_, event_dispatcher())
+        .WillOnce(ReturnRef(event_dispatcher_));
+    hmi_language_handler_ =
+        ::utils::MakeShared<am::HMILanguageHandler>(app_manager_);
+  }
+
+  void InitHMIAtciveLanguage(hmi_apis::Common_Language::eType ui_language,
+                             hmi_apis::Common_Language::eType vr_language,
+                             hmi_apis::Common_Language::eType tts_language) {
+    EXPECT_CALL(hmi_capabilities_, active_ui_language())
+        .WillRepeatedly(Return(ui_language));
+    EXPECT_CALL(hmi_capabilities_, active_vr_language())
+        .WillRepeatedly(Return(vr_language));
+    EXPECT_CALL(hmi_capabilities_, active_tts_language())
+        .WillRepeatedly(Return(tts_language));
+  }
+
+  void Init(hmi_apis::Common_Language::eType ui_language,
+            hmi_apis::Common_Language::eType vr_language,
+            hmi_apis::Common_Language::eType tts_language) {
+    InitHMIAtciveLanguage(ui_language, vr_language, tts_language);
+    hmi_language_handler_->set_default_capabilities_languages(
+        ui_language, vr_language, tts_language);
+  }
+
+  ApplicationSharedPtr CreateMockApp(const uint32_t app_id,
+                                     bool expect_call = false) const {
+    ApplicationSharedPtr app = ::utils::MakeShared<MockApp>();
+    if (expect_call) {
+      EXPECT_CALL(*app, app_id()).WillRepeatedly(Return(app_id));
+    } else {
+      ON_CALL(*app, app_id()).WillByDefault(Return(app_id));
+    }
+    return app;
+  }
+
+  void AddMockAppsToApplicationSet(ApplicationSet& app_set,
+                                   const uint32_t start_app_id,
+                                   uint32_t count,
+                                   bool expect_call = false) {
+    for (; count > 0; --count) {
```

Comment:
@LitvinenkoIra If you will use while(count --> 0) will not forget change line  
app_set.insert(CreateMockApp(start_app_id + (count - 1), expect_call)); 
to app_set.insert(CreateMockApp(start_app_id + count, expect_call));